grizli/prep.py

```diff
@@ -5595,3 +5797,26 @@ def extract_fits_log(file='idk106ckq_flt.fits', get_dq=True):
                 log['DQv{0}'.format(chip)] = list(dq[mask].astype(str))
 
     return log
+
+
+def check_isJWST(infile=''):
+    '''
+    Check if a file or list of files is JWST image. 
+    Parameters
+    ----------
+    infile : fits file
+        File to test
+    
+    Returns
+    -------
+    isJWST : bool
+    '''
+
+    hdu = pyfits.open(infile)
+    try:
+        isJWST = (hdu[0].header['TELESCOP'] == 'JWST')
+        hdu.close()
+    except:
```

Comment:
what exception is raised here?  If `TELESCOP` not found?  If so, make the test explicit:
```python
with pyfits.open(infile) as hdu:
    if 'TELESCOP' in hdu[0].header:
        isJWST = (hdu[0].header['TELESCOP'] == 'JWST')
    else:
        isJWST = False
```

And it also occurs to me that `isJWST` throughout should probably just be handled rather by `INSTRUMENT`, since, for example, you have a different array shape for MIRI.

So rather, you just pull out `INSTRUMENT` and then various tests would be 
changed from `if isJWST` to `if _instrument in ['NIRISS','NIRCAM']`